python_modules/dagster-airflow/dagster_airflow/dagster_plugin.py

```diff
@@ -454,7 +486,17 @@ def execute(self, context):
             self._run_id = context['dag_run'].run_id
         try:
             # FIXME implement intermediate result persistence to S3
-            return super(DagsterOperator, self).execute(context)
+
+            self.log.debug('Executing with query: {query}'.format(query=self.query))
+            raw_res = super(DagsterOperator, self).execute(context)
+            res = json.loads(raw_res)
+            if res['data']['startSubplanExecution']['hasFailures']:
+                raise AirflowException(
```

Comment:
shouldn't we be searching for the step that failed? there's no guarantee that it is first step no?

Reply:
if there is only one step, i think adding a check.invariant would be good for documentation

Reply:
yep, this is improved in the immediately following PR